src/base/bittorrent/trackerentry.cpp

```diff
@@ -30,11 +30,59 @@
 
 #include <algorithm>
 
+#include <QCoreApplication>
 #include <QString>
+#include <QStringList>
 #include <QUrl>
+#include <QVector>
 
 using namespace BitTorrent;
 
+TrackerEntry::Endpoint::Endpoint(const lt::announce_endpoint &nativeEndpoint)
+    : m_nativeEndpoint(nativeEndpoint)
+{
+}
+
+QString TrackerEntry::Endpoint::name() const
+{
+    std::stringstream s;
+    s << m_nativeEndpoint.local_endpoint;
+    return QString::fromStdString(s.str());
+}
+
+int TrackerEntry::Endpoint::numSeeds() const
+{
+    return m_nativeEndpoint.scrape_complete;
+}
+
+int TrackerEntry::Endpoint::numLeeches() const
+{
+    return m_nativeEndpoint.scrape_incomplete;
+}
+
+int TrackerEntry::Endpoint::numDownloaded() const
+{
+    return m_nativeEndpoint.scrape_downloaded;
+}
+
+QString TrackerEntry::Endpoint::message() const
+{
+    if (!m_nativeEndpoint.message.empty())
```

Comment:
Can't there be messages here when it's not working (if it was working before)?

Reply:
LT updates this message whether it failed or succeeded to contact tracker. So it will be updated or empty.